src/inmanta/deploy/scheduler.py

```diff
@@ -628,6 +628,18 @@ async def report_resource_state(
                             deploying=set(),
                         )
 
+    def get_last_non_deploying_state_for_dependencies(
+        self, resource_id: ResourceIdStr
+    ) -> dict[ResourceIdStr, const.ResourceState]:
+        """
+        Get resource state for every dependency of a give resource from the scheduler state
+
+        :param resource_id: The id of the resource to find the dependencies for
+        """
+        provides_view: Mapping[ResourceIdStr, Set[ResourceIdStr]] = self._state.requires
```

Comment:
I think this variable name is confusing since these are the resources that this one requires and not the other way around (I think ?)